sensor/DigitalFenceSensor.js

```diff
@@ -0,0 +1,193 @@
+/*    Copyright 2016-2020 Firewalla Inc.
+ *
+ *    This program is free software: you can redistribute it and/or  modify
+ *    it under the terms of the GNU Affero General Public License, version 3,
+ *    as published by the Free Software Foundation.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+'use strict';
+
+const log = require('../net2/logger.js')(__filename);
+
+const Sensor = require('./Sensor.js').Sensor;
+
+const cp = require('child_process');
+const spawn = cp.spawn;
+const util = require('util');
+const execAsync = util.promisify(cp.exec);
+const f = require('../net2/Firewalla.js');
+const extensionManager = require('./ExtensionManager.js')
+const FireRouter = require('../net2/FireRouter.js');
+
+const wifiKeyPrefix = "digitalfence:wifi:";
+const bluetoothKey = "digitalfence:bluetooth:";
+const wifiCache = {}
+const featureName = "digitalfence"
+
+const rclient = require('../util/redis_manager.js').getRedisClient()
+
+class DigitalFenceSensor extends Sensor {
+
+  async apiRun() {
+
+    extensionManager.onGet("digitalfence", async () => {
+      const wifiRst = {};
+      const bluetoothRst = {};
+      const keys = await rclient.keysAsync("digitalfence:*");
+      await Promise.all(keys.map( async key => {
+        const value = await rclient.hgetallAsync(key);
+        if (key.includes("wifi")) {
+          wifiRst[key] = value;
+        }
+        if (key.includes("bluetooth")) {
+          bluetoothRst[key] = value;
+        }
+      }))
+      return {"wifi": wifiRst, "bluetooth": bluetoothRst};
+    });
+  }
+
+  run() {
+    this.hookFeature(featureName);
```

Comment:
Need to hook MSG_SYS_NETWORK_INFO_RELOADED in case network config is changed in the future. So as to use a new interface according to the latest configuration.